File: tests/rosetta/transpiler/php/find-common-directory-path.php
<?php
ini_set('memory_limit', '-1');
$now_seed = 0;
$now_seeded = false;
$s = getenv('MOCHI_NOW_SEED');
if ($s !== false && $s !== '') {
    $now_seed = intval($s);
    $now_seeded = true;
}
function _now() {
    global $now_seed, $now_seeded;
    if ($now_seeded) {
        $now_seed = ($now_seed * 1664525 + 1013904223) % 2147483647;
        return $now_seed;
    }
    return hrtime(true);
}
$__start_mem = memory_get_usage();
$__start = _now();
  function splitPath($p) {
  $parts = [];
  $cur = '';
  $i = 0;
  while ($i < strlen($p)) {
  if (substr($p, $i, $i + 1 - $i) == '/') {
  if ($cur != '') {
  $parts = array_merge($parts, [$cur]);
  $cur = '';
};
} else {
  $cur = $cur . substr($p, $i, $i + 1 - $i);
}
  $i = $i + 1;
};
  if ($cur != '') {
  $parts = array_merge($parts, [$cur]);
}
  return $parts;
};
  function joinPath($parts) {
  $s = '';
  $i = 0;
  while ($i < count($parts)) {
  $s = $s . '/' . $parts[$i];
  $i = $i + 1;
};
  return $s;
};
  function commonPrefix($paths) {
  if (count($paths) == 0) {
  return '';
}
  $base = splitPath($paths[0]);
  $i = 0;
  $prefix = [];
  while ($i < count($base)) {
  $comp = $base[$i];
  $ok = true;
  foreach ($paths as $p) {
  $parts = splitPath($p);
  if ($i >= count($parts) || $parts[$i] != $comp) {
  $ok = false;
  break;
}
};
  if ($ok) {
  $prefix = array_merge($prefix, [$comp]);
} else {
  break;
}
  $i = $i + 1;
};
  return joinPath($prefix);
};
  function main() {
  $paths = ['/home/user1/tmp/coverage/test', '/home/user1/tmp/covert/operator', '/home/user1/tmp/coven/members', '/home//user1/tmp/coventry', '/home/user1/././tmp/covertly/foo', '/home/bob/../user1/tmp/coved/bar'];
  $c = commonPrefix($paths);
  if ($c == '') {
  echo rtrim('No common path'), PHP_EOL;
} else {
  echo rtrim('Common path: ' . $c), PHP_EOL;
}
};
  main();
$__end = _now();
$__end_mem = memory_get_usage();
$__duration = max(1, intdiv($__end - $__start, 1000));
$__mem_diff = max(0, $__end_mem - $__start_mem);
$__bench = ["duration_us" => $__duration, "memory_bytes" => $__mem_diff, "name" => "main"];
$__j = json_encode($__bench, 128);
$__j = str_replace("    ", "  ", $__j);
echo $__j, PHP_EOL;

File: tests/rosetta/transpiler/Java/find-common-directory-path.java
public class Main {

    static String[] splitPath(String p) {
        String[] parts = new String[]{};
        String cur = "";
        int i = 0;
        while (i < _runeLen(p)) {
            if ((_substr(p, i, i + 1).equals("/"))) {
                if (!(cur.equals(""))) {
                    parts = java.util.stream.Stream.concat(java.util.Arrays.stream(parts), java.util.stream.Stream.of(cur)).toArray(String[]::new);
                    cur = "";
                }
            } else {
                cur = cur + _substr(p, i, i + 1);
            }
            i = i + 1;
        }
        if (!(cur.equals(""))) {
            parts = java.util.stream.Stream.concat(java.util.Arrays.stream(parts), java.util.stream.Stream.of(cur)).toArray(String[]::new);
        }
        return parts;
    }

    static String joinPath(String[] parts) {
        String s = "";
        int i_1 = 0;
        while (i_1 < parts.length) {
            s = s + "/" + parts[i_1];
            i_1 = i_1 + 1;
        }
        return s;
    }

    static String commonPrefix(String[] paths) {
        if (paths.length == 0) {
            return "";
        }
        String[] base = splitPath(paths[0]);
        int i_2 = 0;
        String[] prefix = new String[]{};
        while (i_2 < base.length) {
            String comp = base[i_2];
            boolean ok = true;
            for (String p : paths) {
                String[] parts_1 = splitPath(p);
                if (i_2 >= parts_1.length || !(parts_1[i_2].equals(comp))) {
                    ok = false;
                    break;
                }
            }
            if (ok) {
                prefix = java.util.stream.Stream.concat(java.util.Arrays.stream(prefix), java.util.stream.Stream.of(comp)).toArray(String[]::new);
            } else {
                break;
            }
            i_2 = i_2 + 1;
        }
        return joinPath(prefix);
    }

    static void main() {
        String[] paths = new String[]{"/home/user1/tmp/coverage/test", "/home/user1/tmp/covert/operator", "/home/user1/tmp/coven/members", "/home//user1/tmp/coventry", "/home/user1/././tmp/covertly/foo", "/home/bob/../user1/tmp/coved/bar"};
        String c = String.valueOf(commonPrefix(paths));
        if ((c.equals(""))) {
            System.out.println("No common path");
        } else {
            System.out.println("Common path: " + c);
        }
    }
    public static void main(String[] args) {
        {
            long _benchStart = _now();
            long _benchMem = _mem();
            main();
            long _benchDuration = _now() - _benchStart;
            long _benchMemory = _mem() - _benchMem;
            System.out.println("{");
            System.out.println("  \"duration_us\": " + _benchDuration + ",");
            System.out.println("  \"memory_bytes\": " + _benchMemory + ",");
            System.out.println("  \"name\": \"main\"");
            System.out.println("}");
            return;
        }
    }

    static boolean _nowSeeded = false;
    static int _nowSeed;
    static int _now() {
        if (!_nowSeeded) {
            String s = System.getenv("MOCHI_NOW_SEED");
            if (s != null && !s.isEmpty()) {
                try { _nowSeed = Integer.parseInt(s); _nowSeeded = true; } catch (Exception e) {}
            }
        }
        if (_nowSeeded) {
            _nowSeed = (int)((_nowSeed * 1664525L + 1013904223) % 2147483647);
            return _nowSeed;
        }
        return (int)(System.nanoTime() / 1000);
    }

    static long _mem() {
        Runtime rt = Runtime.getRuntime();
        rt.gc();
        return rt.totalMemory() - rt.freeMemory();
    }

    static int _runeLen(String s) {
        return s.codePointCount(0, s.length());
    }

    static String _substr(String s, int i, int j) {
        int start = s.offsetByCodePoints(0, i);
        int end = s.offsetByCodePoints(0, j);
        return s.substring(start, end);
    }
}
